src/d3a/d3a_core/sim_results/device_statistics.py

```diff
@@ -114,8 +114,10 @@ def calculate_stats_for_infinite_bus(cls, area_dict, subdict, core_stats, curren
         cls._calc_min_max_from_sim_dict(subdict, bought_key_name)
 
     @classmethod
-    def _pv_production_stats(cls, area_dict: Dict, subdict: Dict, core_stats={},
+    def _pv_production_stats(cls, area_dict: Dict, subdict: Dict, core_stats=None,
                              current_market_slot=None):
+        if core_stats is None:
+            core_stats = {}
```

Comment:
Is this a good practice as it is bad ti have an empty dict as default value or what is the rationale behind that?

Reply:
Answered to @faizan2590 above too, https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments